### ISSUE TITLE
Question

### ISSUE THREAD

**Issue by rusyaas** [loopuleasa/Dota2-TheCore-Config-Engine#152]
Is it possible to command illu without even selecting it using autoexec config

**Comment by AetherV** [comment]
I don't quite understand. Are you referring to control groups? Or do you mean the league of legends style micro where you press alt to give commands to a unit (like an illusion) and then automatically reselect your main hero?

**Comment by rusyaas** [comment]
Not like control group. In control group you still have the unit in selection to issue command. I'm not sure about LoL micro. maybe like that? 

eg Currently only main naga is under my selection. When I press mouse5 + aclick the command goes to illusion, but main naga is still under selection.

**Comment by AetherV** [comment]
Can you tell me what you bound mouse5 as? That might help me identify the command (forgot it so yeah). 
Type in: 'bind mouse5'

PS: You are aware that, iirc, it gives a command to all illusions and not just 1.

**Comment by rusyaas** [comment]
eh mouse5 just for example hehe.

**Comment by AetherV** [comment]
So something like this?
https://www.youtube.com/watch?v=B-iopnZlL2o

**Comment by rusyaas** [comment]
Ya, but looks like it command all illusions. Cant it be done on one illu? Like that one comment

"Can i ask you something about?
I wonder is there any command or binding for order to your add (for an example invoker's forge spirits) without select it. CTRL does something like that i mean you can control your add with CTRL without selecting them but you have to move them with invoker in this case. The problem is i can't give an independent order  to my forge spirits whith out selecting them  :( i was thinking to bind something like "holding shift+rightclick to  only to my forge sprits go there or attack but couldn't find... if you help i will be grateful :)﻿"

Seems like the OP said it might be possible through ingame console?

**Comment by dodekeract** [comment]
Couldn't you just bind to keydown "select all other units" and to keyup "select hero"?

**Comment by AetherV** [comment]
Yeah, I am not sure that exists. Perhaps check the [ Manta Configuration](https://manta.dodekeract.report) if it has anything. 

Alternatively, I have been trying to write a little variation of the control group micro, based on quick-cast. The basic idea is to be able to select a unit with say 1, issue commands while holding 1 and then automatically reselect your main hero. Not sure if this is what you want, but it's the closest I can think of.

**Comment by dodekeract** [comment]
@AetherV Manta currently doesn't have that, but I'm thinking about implementing it, since it seems to be something I'd find useful.

**Comment by AetherV** [comment]
Yeah, I am thinking of doing that too, been playing way too much Naga Siren recently. Got a few concepts that should work. 

If you want to, I can give you the code once I finish writing it (and probably mess up the pull request knowing me).

**Comment by dodekeract** [comment]
@AetherV yeah, would be helpful, so I don't have to look that deeply into it. 🙂

**Comment by rusyaas** [comment]
Ya automatically reselect main unit is ok too. OO i see it's still under development. Wish you all the best to complete it.

**Comment by AetherV** [comment]
Well congrats @rusyaas, you successfully requested a new feature :+1: 

@dodekeract hopefully I can get a draft done after tomorrow. Got to get back to learning today. Yay for Informatik exam tomorrow. . .

**Comment by dodekeract** [comment]
@AetherV haha, my semester just started today, studying computer science too

**Comment by AetherV** [comment]
Right, @rusyaas. I have an initial version of this, however I am probably going to add some extra features just cause I got some free time right now. 
You can download this at https://github.com/AetherV/Dota2-TheCore-Config-Engine/tree/Quickcast-Micro, however it may have a few bugs. 

@dodekeract Check the pull request if you want to have the code. I am working on a toggle system for the 2 micro systems, hopefully that will be out soon. 

I am also in the progress of coding a third one that I personally find more useful, namely having you select the unit using a key and then being able to press that key again to get back to your main hero, although there may be some limitations regarding the config system, so there are bound to be bugs.

**Comment by dodekeract** [comment]
@AetherV is your third variation different from a manta 2-cycle between "select unit" and "select hero"?

**Comment by AetherV** [comment]
I'm not quite sure. I am getting slightly confused with where to look for the code. Mind giving me a link to the relevant bits (can't javascript unfortunately). From what I can garner from the description, it doe not seem to be there, but I am not quite sure. 

The basic idea is:

say the hotkey for ctrl group 1 is "2"

1. Have Hero Selected
2. Press 2
3. Select (optionally center at unit) units in ctrl group 1
4. Press 2 again
5. Selects and centers at main hero

**Comment by dodekeract** [comment]
@AetherV for the new features [this](https://github.com/dodekeract/manta-config-engine/blob/196bf2bb5bde1bbf70e77f8ba92fa05cfcf460c9/source/parser/dependency.js#L121-198) is as close as it gets to autoexec.

From your description this sounds like a 2-cycle containing "select hero" and "select unit".

**Comment by AetherV** [comment]
I'll be honest, I have no clue how that script works, probably cause my head is trying to read everything as Java.
I'll take a shot at programming it myself. When it's done, I'll go add a note over at the Manta Config Engine so you can compare them.

**Comment by loopuleasa** [comment]
So, a feature for holding down a button, having illusions selected, then releasing it having the hero selected?

Could be useful, but I am unsure on what button to fit it.

**Comment by dodekeract** [comment]
@loopuleasa, this is already in Manta, I implemented like 5 different variations of this and it's documented on the website, when you try to bind a key to select you should get a help text. I'd suggest you to look at what all of them are about in that help text, since I personally replaced all of my control groups with one of these new settings.
